src/UI/Tooltip.elm

```diff
@@ -1,47 +1,106 @@
-module UI.Tooltip exposing (Arrow(..), Tooltip, tooltip, view, withArrow)
-
-import Html exposing (Html, div, span)
+module UI.Tooltip exposing
+    ( Arrow(..)
+    , Content(..)
+    , MenuItem
+    , Tooltip
+    , tooltip
+    , view
+    , withArrow
+    , withPosition
+    )
+
+import Html exposing (Html, a, div, span, text)
 import Html.Attributes exposing (class)
-
-
-type Arrow
-    = None
-    | Top
-    | TopLeft
-    | TopRight
-    | Right
-    | RightTop
-    | RightBottom
-    | Bottom
-    | BottomLeft
-    | BottomRight
-    | Left
-    | LeftTop
-    | LeftBottom
+import Html.Events exposing (onClick)
+import UI.Icon as Icon exposing (Icon)
 
 
 type alias Tooltip msg =
     { arrow : Arrow
     , trigger : Html msg
-    , content : Html msg
+    , content : Content msg
+    , position : Position
     }
 
 
-tooltip : Html msg -> Html msg -> Tooltip msg
+type Arrow
+    = None
+    | Start
+    | Middle
+    | End
+
+
+{-| Position relative to trigger
+-}
+type Position
+    = Above
+    | Below
+    | LeftOf
+    | RightOf
+
+
+type alias MenuItem msg =
+    { icon : Icon msg, label : String, onClick : msg }
+
+
+type Content msg
+    = Text String
+    | Rich (Html msg)
+    | Menu (List (MenuItem msg))
+
+
+tooltip : Html msg -> Content msg -> Tooltip msg
 tooltip trigger content =
-    { arrow = Top, trigger = trigger, content = content }
+    { arrow = Middle
+    , trigger = trigger
+    , content = content
+    , position = Below
+    }
 
 
 withArrow : Arrow -> Tooltip msg -> Tooltip msg
 withArrow arrow tooltip_ =
     { tooltip_ | arrow = arrow }
 
 
+withPosition : Position -> Tooltip msg -> Tooltip msg
+withPosition pos tooltip_ =
+    { tooltip_ | position = pos }
+
+
 view : Tooltip msg -> Html msg
-view { arrow, content, trigger } =
+view { arrow, content, trigger, position } =
     let
+        viewMenuItem item =
+            a [ class "tooltip-menu-item", onClick item.onClick ]
+                [ Icon.view item.icon, text item.label ]
+
+        content_ =
+            case content of
+                Text t ->
+                    text t
+
+                Rich html ->
+                    html
+
+                Menu items ->
+                    div [ class "tooltip-menu-items" ] (List.map viewMenuItem items)
+
         tooltip_ =
-            div [ class "tooltip", class (arrowToClass arrow) ] [ content ]
+            -- The tooltip includes a small bridge (made with padding) above
+            -- the bubble to allow the user to hover into the tooltip and click
+            -- links etc.
```

Comment:
This "bridge" work fixed a bug where people couldn't click links inside tooltips; especially exercised in docs.